oculus_ros2/scripts/display_oculus_file.py

```diff
@@ -0,0 +1,192 @@
+    # print("temperature",         msg.temperature())
+    # print("pressure",            msg.pressure())
+
+    bearings = 0.01 * np.array(msg.bearing_data())
+    linearAngles = np.linspace(bearings[0], bearings[-1], len(bearings))
+    rawPingData = np.array(msg.raw_ping_data())
+    gains = np.ones(
+        [
+            msg.range_count(),
+        ],
+        dtype=np.float32,
+    )
+    if msg.has_gains():
+        gains = np.array(msg.gains())
+    pingData = np.array(msg.ping_data()) / np.sqrt(gains)[:, np.newaxis]
+    # print("pingData =", list(pingData))
+    # print('has gains   :', msg.has_gains())
+    # print('sample size :', msg.sample_size())
+    # print('ping shape  :', pingData.shape)
+
+    # _, ax = plt.subplots(1,1,)
+    # ax.plot(bearings,     '-o', label='bearings')
+    # ax.plot(linearAngles, '-o', label='linear bearings')
+    # ax.grid()
+    # ax.legend()
+    # ax.set_xlabel('bearing index')
+    # ax.set_ylabel('bearing angle')
+
+    # _, ax = plt.subplots(1,1)
+    # ax.plot(gains, '-o', label='gains')
+    # ax.grid()
+    # ax.legend()
+    # ax.set_xlabel('range index')
+    # ax.set_ylabel('range gain')
+
+    ax[0].imshow(rawPingData)
+    ax[0].set_ylabel("Range index")
+    ax[0].set_xlabel("Bearing index")
+    ax[0].set_title("Raw ping data")
+
+    ax[1].imshow(pingData)
+    ax[1].set_xlabel("Bearing index")
+    ax[1].set_title("Ping data rescaled with gains")
+
+    plt.pause(args.rate)
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        prog="OculusFileReader",
+        description="Example of how to read and display the content of a .oculus "
+        + "file. This will display the first ping from a the file.",
+    )
+    parser.add_argument("filename", type=str, help="Path to a .oculus file to display")
+    parser.add_argument(
+        "-r", "--rate", type=float, default=1e-2, help=". Default to 1e-2"
+    )
+
+    args = parser.parse_args()
+
+    print("Opening", args.filename)
+    file = OculusFileReader(args.filename)
+    # this can be called several time to iterate through the pings
+    oculus_ping_msg = file.read_next_ping()
+    # will return None when finished
+    if oculus_ping_msg is None:
+        print("[oculus_to_bag] File seems to be empty. Aborting.")
+        return
+    # print("type(oculus_ping_msg) =", type(oculus_ping_msg))
+
+    k = 0
+    start_time = time.perf_counter()
+    _, ax = plt.subplots(1, 2)
+    while oculus_ping_msg:
+        k += 1
+        if not k % 20:
+            # print(k)
+            elapsed_time = time.perf_counter() - start_time
+            print(
+                "[oculus_to_bag] {} pings have been red in {:.2f} seconds.".format(
+                    k, elapsed_time
+                ),
+                end="\r",
+            )
+
+            display_oculus_ping(oculus_ping_msg, ax, args)
+
+        oculus_ping_msg = file.read_next_ping()
+    print("")
+
+
+if __name__ == "__main__":
+
+    main()
```

Comment:
Why putting all the main code in a main function ?

Reply:
To put
```python
   parser = argparse.ArgumentParser(
        prog="OculusFileReader",
        description="Example of how to read and display the content of a .oculus "
        + "file. This will display the first ping from a the file.",
    )
    parser.add_argument("filename", type=str, help="Path to a .oculus file to display")
    parser.add_argument(
        "-r", "--rate", type=float, default=1e-2, help=". Default to 1e-2"
    )

    args = parser.parse_args()
    ```
    outside of it.